test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -233,6 +233,42 @@ public void GetMetadataForProperties_SetsContainerAsExpected()
             Assert.Same(model, metadata[2].Container);
         }
 
+        [Fact]
+        public void GetMetadataForPropertiesFromModelMetadataType()
+        {
+            // Arrange
+            var model = new PropertyModel { LocalAttributes = 8, MetadataAttributes = "hello", MixedAttributes = 21.12 };
+            var provider = new TestableAssociatedMetadataProvider();
+
+            // Act
+            // Call ToList() to force the lazy evaluation to evaluate
+            provider.GetMetadataForProperties(model, typeof(PropertyViewModel)).ToList();
+
+            // Assert
+            var local = Assert.Single(
+                provider.CreateMetadataPrototypeLog,
+                m => m.ContainerType == typeof(PropertyModel) && m.PropertyName == "LocalAttributes");
+            Assert.Equal(typeof(int), local.ModelType);
+            Assert.True(local.Attributes.Any(a => a is RequiredAttribute));
+            Assert.True(local.Attributes.Any(a => a is RangeAttribute));
```

Comment:
avoid `Assert.True()` in new tests: use `Assert.Single(local.Attributes.OfType<RequiredAttribute>())` and `Assert.Contains(local.Attributes.OfType<RangeAttribute>())`.

existing tests in this class don't follow this practise but leave them alone in this PR.